tests/integration/bundle/kafka_helpers.py

```diff
@@ -81,25 +88,132 @@ def show_unit(unit_name: str, model_full_name: str) -> Any:
     return yaml.safe_load(result)
 
 
-def get_zookeeper_connection(unit_name: str, model_full_name: str) -> Tuple[List[str], str]:
-    result = show_unit(unit_name=unit_name, model_full_name=model_full_name)
+def get_secret_by_label(model_full_name: str, label: str, owner: str) -> dict[str, str]:
```

Comment:
I have updated this implementation, since in the [previous implementation](https://github.com/canonical/kafka-operator/blob/main/tests/integration/helpers.py#L310) I found an edge case that when there was no secret matching with owner and label, the data returned was the last `secret_id` (given the previous for loop-continue-break pattern - which I found it a bit of a weird pattern TBH)